cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py

```diff
@@ -37,3 +37,8 @@ def create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell
             )
 
     return sample_lane_sequencing_metrics
+
+
+def get_flow_cell_name_from_directory(flow_cell_dir: Path) -> str:
+    """Get the flow cell name from the directory name."""
+    return flow_cell_dir.name.split("_")[-1][1:]
```

Comment:
I think this should be a property of the flow cell directory. It already has an implemented property called `split_flow_cell_name`, you could also use that to compile with DRY. In summary I would suggest either using:
1. flow_cell_dir.split_flow_cell_name[0]
2. create a new property in FlowCellDirectory class that outputs this

Reply:
I removed the function and added a flow cell name attribute to the flow cell directory data class. Furthermore, I renamed some attributes and functions that make it clearer that a flow cell directory name is being parsed from which we then extract the flow cell name